File: Habitica/src/main/java/com/magicmicky/habitrpgwrapper/lib/models/responses/HabitResponse.java
package com.magicmicky.habitrpgwrapper.lib.models.responses;

import com.magicmicky.habitrpgwrapper.lib.models.Notification;

import java.util.List;

/**
 * Created by krh12 on 11/23/2016.
 */

public class HabitResponse<T> {

    private Boolean success;
    public T data;
    public List<Notification> notifications;

    /**
     *
     * @return
     * The success
     */
    public Boolean getSuccess() {
        return success;
    }

    /**
     *
     * @param success
     * The success
     */
    public void setSuccess(Boolean success) {
        this.success = success;
    }

    /**
     *
     * @return
     * The data
     */
    public T getData() {
        return data;
    }
}

File: common/src/main/java/com/habitrpg/common/habitica/models/responses/HabitResponse.kt
package com.habitrpg.common.habitica.models.responses

import com.habitrpg.common.habitica.models.Notification

class HabitResponse<T> {
    var data: T? = null
    var notifications: List<Notification>? = null
    var success: Boolean? = null
    var message: String? = null
}
